--- zabbix-2.2.2/src/zabbix_agent/logfiles.c.orig       2014-02-12 18:04:37.000000000 +0900
+++ zabbix-2.2.2/src/zabbix_agent/logfiles.c    2014-04-22 02:36:09.150429544 +0900
@@ -21,6 +21,7 @@
 #include "logfiles.h"
 #include "log.h"
 #include "zbxregexp.h"
+#include <string.h>

 /******************************************************************************
  *                                                                            *
@@ -617,6 +618,7 @@
        struct stat     buf;
        int             nbytes, ret = FAIL;
        char            buffer[MAX_BUFFER_LEN];
+       char *matchtag = "test:";

        assert(NULL != filename);
        assert(NULL != lastlogsize);
@@ -661,8 +663,19 @@
                        {
                                *lastlogsize += nbytes;
                                *value = convert_to_utf8(buffer, nbytes, encoding);
+                               zabbix_log(LOG_LEVEL_DEBUG, "logfile.c::process_log:%s" ,*value);
                                zbx_rtrim(*value, "\r\n ");
                        }
+
+                        if(*value != NULL) {
+                            if(strstr(*value,matchtag) != NULL) {
+                               zabbix_log(LOG_LEVEL_DEBUG, "match: logfile.c::process_log:%s:(%s)" ,*value,matchtag);
+                            }else{
+                               zabbix_log(LOG_LEVEL_DEBUG, "not match: logfile.c::process_log:%s->%s" ,*value,NULL);
+                                *value = NULL;
+                            }
+                        }
+
                        ret = SUCCEED;
                }
                else
